### ISSUE TITLE
[Suggestion] Work as a Unity Package (using the Unity Package Manager)

### ISSUE THREAD

**Issue by zeh** [vladfaust/unity-wakatime#5]
Unity is moving towards a "package" architecture with UPM, the Unity Package Manager. Now, features can be "installed" in projects and be managed externally, as packages, without polluting the game's code base. While this is on early days, the intention is for this to work similarly to NPM packages: that is, you should be able to install third-party dependencies at will. Unity 2018.1 is the first one to feature UPM.

WakaTime support is an ideal target for that, in my opinion. Pasting WakaTime-specific code into a game codebase just so you can get stats seems kind of icky.

Here's an introduction to the changes:
https://blogs.unity3d.com/2018/05/04/project-management-is-evolving-unity-package-manager-overview/
https://docs.unity3d.com/Packages/com.unity.package-manager-ui@1.8/manual/index.html
https://forum.unity.com/forums/package-manager.150/

_Currently_, only official (Unity-created) packages are allowed, while they're moving official Unity features into packages. However, the intention is to support packages by simply "installing" them from a Git URL in the future.

In the meantime, we can "install" packages by copying them locally and referring to them by their relative path location. They would be _out_ of the project's folder.

Here's instructions on how to do that:
https://gist.github.com/LotteMakesStuff/6e02e0ea303030517a071a1c81eb016e

This should serve as a great preparation for a Git UPM package for WakaTime.

## TL;DR:

In the future, developers will be able to use WakaTime inside Unity by just pasting one line of code into their `manifest.json`, rather than downloading scripts and placing them inside the project. Making this available should probably be a priority for WakaTime Unity plugin maintainers.

## Plan

Since this a "pure" extension that is 100% based on C#, I can help with that with PRs _in case it's not already in the roadmap_. My plan is this:

1. Document how to install locally, but on a separate folder, UPM style (since this already works). It will requires some changes to the codebase (adding a `package.json`).
2. Document how to install using a Git URL, once that feature is available. It will _probably_ require changes to the codebase structure, depending on whether this upcoming Git support will be able to target a folder or not.

Let me know if this plan is OK or otherwise if you're already working on it.

**Comment by vladfaust** [comment]
Oh, thanks for such a well-structured text! 

I didn't have this feature on my roadmap and I'll be very happy if you implement this feature due to I agree that packages are essentially better than copy-pasting.
